src/components/flowchart-primary-toolbar/flowchart-primary-toolbar.js

```diff
@@ -124,9 +130,12 @@ export const mapDispatchToProps = (dispatch) => ({
   onToggleTextLabels: (value) => {
     dispatch(toggleTextLabels(Boolean(value)));
   },
+  onToggleExpandPipelines: (ids) => {
+    dispatch(toggleModularPipelinesExpanded(ids));
+  },
   onToggleExpandAllPipelines: (isExpanded) => {
     dispatch(toggleExpandAllPipelines(isExpanded));
-    dispatch(loadInitialPipelineData());
+    dispatch(toggleAllModularPipelinesExpanded(isExpanded));
```

Comment:
If we decide to go with this refactor, I suggest renaming two reducers to improve clarity: TOGGLE_EXPAND_ALL_PIPELINES (which returns a boolean) and TOGGLE_ALL_MODULAR_PIPELINES_EXPANDED (which updates the state). Currently, their names are very similar, even though they do slightly different things.